gst/mqtt/mqttsink.c

```diff
@@ -816,10 +819,8 @@ gst_mqtt_sink_event (GstBaseSink * basesink, GstEvent * event)
 
   switch (type) {
     case GST_EVENT_EOS:
-      g_mutex_lock (&self->mqtt_sink_mutex);
-      self->mqtt_sink_state = SINK_RENDER_EOS;
+      g_atomic_int_set (&self->mqtt_sink_state, SINK_RENDER_EOS);
       g_cond_broadcast (&self->mqtt_sink_gcond);
```

Comment:
Doesn't cond-broadcast require to have a mutex locked?

Reply:
@myungjoo Right. I will modify it.